src/finn/transformation/fpgadataflow/insert_hook.py

```diff
@@ -82,7 +82,7 @@ def apply(self, model):
                     if n0_hook in list_supported_hooks:
```

Comment:
Related to line 74.

Just listing this here, I think you have a better idea what exactly the CheckSum layer should perform.
The only node that can have multiple outputs is the DuplicateStreams_Batch. What will happen is that this transform will insert 2 CheckSum nodes for 2 identical streams (going out of the DuplicateStreams_Batch), which perhaps is overkill. If the layer preceding the DuplicateStreams_Batch has a CheckSum node, then perhaps we don't need to compute it afterwards again.